superset/utils/async_query_manager.py

```diff
@@ -0,0 +1,200 @@
+        self._stream_limit: Optional[int]
+        self._stream_limit_firehose: Optional[int]
+        self._jwt_cookie_name: str
+        self._jwt_cookie_secure: bool = False
+        self._jwt_secret: str
+
+    def init_app(self, app: Flask) -> None:
+        config = app.config
+        if (
+            config["CACHE_CONFIG"]["CACHE_TYPE"] == "null"
+            or config["DATA_CACHE_CONFIG"]["CACHE_TYPE"] == "null"
+        ):
+            raise Exception(
+                """
+                Cache backends (CACHE_CONFIG, DATA_CACHE_CONFIG) must be configured
+                and non-null in order to enable async queries
+                """
+            )
+
+        if len(config["GLOBAL_ASYNC_QUERIES_JWT_SECRET"]) < 32:
+            raise AsyncQueryTokenException(
+                "Please provide a JWT secret at least 32 bytes long"
+            )
+
+        self._redis = redis.Redis(  # type: ignore
+            **config["GLOBAL_ASYNC_QUERIES_REDIS_CONFIG"], decode_responses=True
+        )
+        self._stream_prefix = config["GLOBAL_ASYNC_QUERIES_REDIS_STREAM_PREFIX"]
+        self._stream_limit = config["GLOBAL_ASYNC_QUERIES_REDIS_STREAM_LIMIT"]
+        self._stream_limit_firehose = config[
+            "GLOBAL_ASYNC_QUERIES_REDIS_STREAM_LIMIT_FIREHOSE"
+        ]
+        self._jwt_cookie_name = config["GLOBAL_ASYNC_QUERIES_JWT_COOKIE_NAME"]
+        self._jwt_cookie_secure = config["GLOBAL_ASYNC_QUERIES_JWT_COOKIE_SECURE"]
+        self._jwt_secret = config["GLOBAL_ASYNC_QUERIES_JWT_SECRET"]
+
+        @app.after_request
+        def validate_session(  # pylint: disable=unused-variable
+            response: Response,
+        ) -> Response:
+            reset_token = False
+            user_id = session["user_id"] if "user_id" in session else None
+
+            if "async_channel_id" not in session or "async_user_id" not in session:
+                reset_token = True
+            elif user_id != session["async_user_id"]:
+                reset_token = True
+
+            if reset_token:
+                async_channel_id = str(uuid.uuid4())
+                session["async_channel_id"] = async_channel_id
+                session["async_user_id"] = user_id
+
+                token = self.generate_jwt(
+                    {"channel": async_channel_id, "user_id": user_id}
+                )
+
+                response.set_cookie(
+                    self._jwt_cookie_name,
+                    value=token,
+                    httponly=True,
+                    secure=self._jwt_cookie_secure,
+                    # max_age=max_age or config.cookie_max_age,
+                    # domain=config.cookie_domain,
+                    # path=config.access_cookie_path,
+                    # samesite=config.cookie_samesite
+                )
+
+            return response
+
+    def generate_jwt(self, data: Dict[str, Any]) -> str:
+        encoded_jwt = jwt.encode(data, self._jwt_secret, algorithm="HS256")
```

Comment:
This algo needs to be configurable. Also, the JWT contents here aren't following the JWT standard. Typically, `sub` is used to track the "id" of a user. Again, the lib mentioned above deals with this.

Reply:
The [JWT standard](https://tools.ietf.org/html/rfc7519) doesn't prescribe any required claim fields, but happy to use the `sub` field for `user_id` here (or just remove it altogether). Also happy to make the algo configurable, but there's a point of diminishing returns for configuration here, IMO.

Reply:
Additional configuration can also be added in future PRs as needs are more clearly defined.

Reply:
Right. The `claim` field is up to the implementor. A typical "bare bones" JWT would look something like:
```
{
  "sub": "1234567890",
  "name": "John Doe",
  "iat": 1516239022,
  "exp": 1516239022,
  "claims": {
     ... whatever stuff you want
   }
}
```

Reply:
> 4.1.  Registered Claim Names
> None of the claims defined below are intended to be mandatory to use or implement in all
   cases, but rather they provide a starting point for a set of useful,
   interoperable claims.  Applications using JWTs should define which
   specific claims they use and when they are required or optional.